components/hab/src/command/pkg/binlink.rs

```diff
@@ -106,6 +118,9 @@ where
             if bin_file.file_type()?.is_dir() {
                 continue;
             }
+            if cfg!(target_os = "windows") && !is_executable(bin_file.path().extension()) {
+                continue;
+            }
```

Comment:
I found the `is_executable` function hard to understand and a little misleading since it's not marked `#[cfg(windows)]`, but it only *works* for windows. I understand it's necessary because it's a separate function, so otherwise you'd have to provide a non-windows implementation, but how about side-stepping the whole issue by doing the check inline:
```rust
            if cfg!(windows) {
                if let Some(executable_extensions) = env::var_os("PATHEXT") {
                    if env::split_paths(&executable_extensions)
                        .any(|e| extensions_match(Path::new(&e), &bin_file.path()))
                    {
                        continue;
                    }
                }
            }
```
Then the factored-out `extensions_match` is platform-agnostic:
```rust
fn extensions_match(p1: &Path, p2: &Path) -> bool {
    match (p1.extension(), p2.extension()) {
        (Some(ext1), Some(ext2)) => {
            ext1.to_string_lossy().to_uppercase() == ext2.to_string_lossy().to_uppercase()
        }
        _ => false,
    }
}
```